src/ModalBase/Portal.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+
+class Portal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.rootName = 'paragon-portal-root';
+    if (typeof document === 'undefined') {
+      this.rootElement = null;
+    } else if (document.getElementById(this.rootName)) {
+      this.rootElement = document.getElementById(this.rootName);
+    } else {
+      const rootElement = document.createElement('div');
+      rootElement.setAttribute('id', this.rootName);
+      this.rootElement = document.body.appendChild(rootElement);
+    }
+  }
+
+  render() {
+    if (this.rootElement) {
+      return ReactDOM.createPortal(
```

Comment:
What does this do?

Reply:
My understanding is that React Portals essentially let you mount a component into a separate DOM node than it would otherwise render within the component tree. E.g., here, any children of `Portal` will actually get rendered as child nodes of `this.rootElement`.

Reply:
What Adam said. And the react docs https://reactjs.org/docs/portals.html.  We use this for the modals because we can't guarantee that no parent element has overflow or z-index set. Portals let us control the [stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context) we render our components in.

Reply:
Cooooool. 
Are there accessibility considerations for using them, or do screen readers just handle them gracefully?

Reply:
We don't get anything like tab or reading order for free through portals. In our case we're using aria-hidden on the app root element and placing focus on the first focusable element of the modal element so no a11y is handled.